pkg/machine/shim/host.go

```diff
@@ -501,22 +502,29 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, opts machine.St
 	// Do not do anything with the system connection if its already
 	// the default system connection.
 	if !conn.Default {
-		// Prompt for system connection update
+		// the user did not explicitly say to update or not
 		if updateSystemConn == nil {
-			response, err := promptUpdateSystemConn()
-			if err != nil {
-				return err
-			}
-			// This might be kind of lame but when using the command, but if you don't
-			// provide some sort of visual cue back to the user, it's unclear what is
-			// going on because the machine startup is going and it looks like things
-			// are frozen
-			if response {
-				fmt.Printf("\nDefault system connection will be changed to %q\n ", connName)
-			} else {
-				fmt.Println("Default system connection will remain unchanged")
+			// Prompt for system connection update if there is a terminal
+			// on stdin
+			// Note: At this point, updateSystemConn is false by declaration
+			// and therefore the conditionals need to stay this way to preserve
+			// that false value.  Do not combine this and the previous conditional.
+			if term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
I am a bit confused about that sentence, I understand what you mean logic wise and it does seem right but it took me more effort to parse the logic now.

WDYT about inverting the if branch? This removes one level of nesting.
```
if updateSystemConn != nil {
    updateDefaultConnection = *updateSystemConn
} else if term.IsTerminal(int(os.Stdin.Fd())) {
   // the user did not explicitly say to update or not and we have a terminal to ask
    prompt code here 
}
```